crates/uv/src/commands/build_frontend.rs

```diff
@@ -780,6 +723,138 @@ async fn build_package(
     Ok(assets)
 }
 
+/// Build a source distribution, either through PEP 517 or through the fast path.
+async fn build_sdist(
+    source_tree: &Path,
+    output_dir: &Path,
+    fast_path: bool,
+    source: &AnnotatedSource<'_>,
+    printer: Printer,
+    message: &str,
+    // Below is only used with PEP 517 builds
+    build_dispatch: &BuildDispatch<'_>,
+    sources: SourceStrategy,
+    dist: Option<&SourceDist>,
+    subdirectory: Option<&Path>,
+    version_id: Option<&str>,
+    build_output: BuildOutput,
+) -> Result<String> {
+    let sdist = if fast_path {
```

Comment:
I might structure these as just two entirely separate methods? That's always my instinct when I have a method that takes a bool and entirely forks behavior on that bool. Sort of a matter of preference, though.